model/converter/thrift/zipkin/to_domain.go

```diff
@@ -178,6 +178,11 @@ func (td toDomain) findServiceNameAndIP(zSpan *zipkincore.Span) (string, int32,
 			return a.Host.ServiceName, a.Host.Ipv4, nil
 		}
 	}
+	for _, a := range zSpan.BinaryAnnotations {
+		if a.Host != nil && a.Host.ServiceName != "" {
+			return a.Host.ServiceName, a.Host.Ipv4, nil
+		}
+	}
```

Comment:
if you're adding this you obviously ran into an issue, but I am curios that kind of binary annotation (aside from LOCAL_COMPONENT already checked for above) could be considered a reliable source of service name/ip? 

NB a comment above this loop would be good explaining.

Reply:
I have added a comment, basically tracer can send a span with just one binary annotation, see also liked issue it happened with correct OT instrumentation and brave-opentracing.